includes/Story_Renderer/HTML.php

```diff
@@ -191,6 +193,49 @@ protected function replace_html_head( $content ) {
 		return $content;
 	}
 
+	/**
+	 * Remove <noscript> AMP boilerplate fragment from the existing markup.
+	 *
+	 * The libxml extension prior to version 2.8.0 has issues parsing <noscript> tags in the <head>.
+	 * That's why we temporarily remove this fragment and re-add it later.
+	 *
+	 * @see https://github.com/ampproject/amp-wp/pull/5097
+	 * @see http://xmlsoft.org/news.html
+	 *
+	 * @param string $content Story markup.
+	 *
+	 * @return string Filtered content.
+	 */
+	protected function remove_noscript_amp_boilerplate( $content ) {
+		return str_replace( '<noscript><style amp-boilerplate="">body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>', '', $content );
+	}
+
+	/**
+	 * Re-add <noscript> AMP boilerplate using DOMDocument.
+	 *
+	 * The libxml extension prior to version 2.8.0 has issues parsing <noscript> tags in the <head>.
+	 * That's why we temporarily remove this fragment and re-add it later.
+	 *
+	 * @see https://github.com/ampproject/amp-wp/pull/5097
+	 * @see http://xmlsoft.org/news.html
+	 *
+	 * @return void
+	 */
+	protected function add_noscript_amp_boilerplate() {
+		$fragment = $this->document->createDocumentFragment();
+		$fragment->appendXml(
+			'<noscript><style amp-boilerplate="">body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>'
+		);
+
+		$head = $this->get_element_by_tag_name( 'head' );
+
+		if ( ! $head ) {
+			return;
+		}
+
+		$head->appendChild( $this->document->importNode( $fragment, true ) );
```

Comment:
Is importing necessary? The fragment was already created in this document, so I think you can just append the fragment as a whole.

Reply:
Maybe not! I'll check